### PR TITLE
Solve #2286. Download imagenet train/val splits alone

### PR DESCRIPTION
**Pull request by ibarrond**
Solving #2286 by allowing individual download of Imagenet2012 TRAIN/VALIDATION splits. This way, the 144GB TRAIN split won't be necessary anymore to evaluate a model on the 7GB VALIDATION split (saving a lot of time and resources).

### CODE REVIEW

**Comment by ibarrond** [comment]
@googlebot I signed it!

File: tensorflow_datasets/image_classification/imagenet.py [hunk 1]
```diff
@@ -187,26 +187,37 @@ def _split_generators(self, dl_manager):
...
+                      'generation anyways to retain backward compatibility.',
+                      train_path)
+    else:
+      splits.append(
+          tfds.core.SplitGenerator(
+              name=tfds.Split.TRAIN,
+              gen_kwargs={
+                  'archive': dl_manager.iter_archive(train_path),
+              }))
+    if not tf.io.gfile.exists(val_path):
```
Comment (Conchylicultor) [inline tensorflow_datasets/image_classification/imagenet.py:208]: Rather than duplicating this code 3 times, how about creating an util function ?

**Review by Conchylicultor — CHANGES_REQUESTED** [review]
Thank you!

**Comment by ibarrond** [comment]
@Conchylicultor Code refactored!

File: tensorflow_datasets/image_classification/imagenet.py [hunk 2]
```diff
@@ -187,42 +187,35 @@ def _split_generators(self, dl_manager):
...
-          tfds.core.SplitGenerator(
-              name=tfds.Split.TEST,
-              gen_kwargs={
-                  'archive': dl_manager.iter_archive(test_path),
-                  'labels_exist': False,
-              }))
+    splits=[]
+    self._add_split_if_exists(splits, tfds.Split.TRAIN, train_path)
+    self._add_split_if_exists(splits, tfds.Split.VALIDATION, val_path,
+                     'validation_labels': self._get_validation_labels(val_path))
```
Comment (Conchylicultor) [inline tensorflow_datasets/image_classification/imagenet.py:198]: This seems to be syntax error. Did you tested this code ?

**Review by Conchylicultor — CHANGES_REQUESTED** [review]

**Review by Conchylicultor — APPROVED** [review]
Thank you!

**Pull request merged**